File: 모듈화/binarysearch.cc
#include<bits/stdc++.h>
using namespace std;
bool bisearch(vector<int> &v, int k){
    int lo =0; int hi= v.size()-1;
    while(lo<=hi){
        int mid = (lo+hi)/2;
        if(v[mid]==k)return 1;
        if(v[mid]<k)lo=mid;
        else hi =mid - 1;
    }   
    return 0;
}
int main(){
    vector<int> v= {2,3,5,6};
    int k;
    cin >> k;
    cout<<bisearch(v,k);
}

File: programmers/day1/binary_search.py
#정렬된 L에서 값 x를 가지는 인덱스를 return 없으면 -1
def solution(L, x):
    lo = 0
    hi = len(L) - 1
    while(lo <= hi) :
        mid = (lo+hi)//2
        if(L[mid]==x):
            return mid
        elif(L[mid]>x):
            hi = mid- 1
        else:
            lo = mid+ 1
    return -1
print(solution([1,2,4],4))